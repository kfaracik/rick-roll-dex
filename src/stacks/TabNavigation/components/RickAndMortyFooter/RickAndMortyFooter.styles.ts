import {StyleSheet} from 'react-native';
import {Colors} from '../../../../shared/utils';

export const styles = StyleSheet.create({
  imageContainer: {
    height: 140,
    backgroundColor: Colors.primary,
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    height: 92,
  },
});
