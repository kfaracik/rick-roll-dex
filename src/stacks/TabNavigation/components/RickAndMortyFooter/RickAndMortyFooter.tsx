import {View, Image} from 'react-native';
import React from 'react';
import rickAndMorty from '../../../../shared/assets/images/rickAndMorty/rickAndMorty@3.webp'; // TODO: automatically select quality
import {styles} from './RickAndMortyFooter.styles';

export const RickAndMortyFooter = () => {
  return (
    <View style={styles.imageContainer}>
      <Image resizeMode="contain" source={rickAndMorty} style={styles.image} />
    </View>
  );
};
