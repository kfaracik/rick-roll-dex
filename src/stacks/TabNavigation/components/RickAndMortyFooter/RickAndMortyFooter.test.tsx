import React from 'react';
import {render} from '@testing-library/react-native';
import {RickAndMortyFooter} from './RickAndMortyFooter';

describe('RickAndMortyFooter component', () => {
  it('should generate snapshot', () => {
    const {toJSON} = render(<RickAndMortyFooter />);
    expect(toJSON()).toMatchSnapshot();
  });
});
