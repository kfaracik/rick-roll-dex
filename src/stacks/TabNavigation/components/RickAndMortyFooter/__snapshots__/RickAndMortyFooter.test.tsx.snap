// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RickAndMortyFooter component should generate snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#224229",
      "height": 140,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../src/shared/assets/images/rickAndMortyBig/rickAndMortyBig.webp",
      }
    }
    style={
      {
        "height": 92,
      }
    }
  />
</View>
`;
